components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2698,8 +2698,17 @@ private void processPatchRequest(List<Patch> patchRequest, IdentityProvider idpT
                                         Constants.ErrorMessage.ERROR_CODE_INVALID_CERTIFICATE_FORMAT, null);
                             }
                         }
+                        if (certificates.contains(value)) {
+                            throw handleException(Response.Status.CONFLICT,
+                                    Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_IDP,
+                                    "Cannot replace certificate as this certificate already exists.");
+                        }
                         certificates.set(index, value);
                         idpToUpdate.setCertificate(base64Encode(StringUtils.join(certificates, "")));
+                    } else if (ArrayUtils.isEmpty(idpToUpdate.getCertificateInfoArray()) ||
```

Comment:
Can `idpToUpdate` be null?

Reply:
ArrayUtils.isEmpty checks whether the given array is empty or null.

Reply:
It's not about the CertificateInfoArray. If the `idpToUpdate` object is null, won't this cause an NPE?

Reply:
It is already handled in https://github.com/wso2/identity-api-server/blob/6b17264995093bc2d93df3fe33ed8c901d688c9e/components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java#L230